torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -231,6 +231,26 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
       return c10::optional<at::Tensor>(updated_val);
     }
     return c10::nullopt;
+  } else if (node->kind() == onnx::Reshape) {
+    assert(inputTensorValues.size() == 2);
+    updated_val = inputTensorValues[0];
+    std::vector<int64_t> shape(inputTensorValues[1].sizes()[0], 0);
+    auto shape_a = inputTensorValues[1].accessor<int64_t, 1>();
+    for (size_t i = 0; i < inputTensorValues[1].sizes()[0]; ++i) {
+      // All shape dim values should be >= -1
+      // onnx::Reshape supports a shape dim value to be zero, in
+      // which case the actual dim value remains unchanged. However,
+      // at::reshape does not support shape dim value to be zero
+      assert(shape_a[i] >= -1);
+      if (shape_a[i] == 0){
+        assert(i < inputTensorValues[0].sizes().size());  // dim with value zero < the dimension size input tensor
```

Comment:
This is a good check. In this particular instance we should have an explicit error message spelling out the issue instead of an assert.